sphinx/testing/util.py

```diff
@@ -244,3 +240,20 @@ def _clean_up_global_state() -> None:
 
     # clean up autodoc global state
     sphinx.pycode.ModuleAnalyzer.cache.clear()
+
+
+_DEPRECATED_OBJECTS: Final[dict[str, tuple[object, str, tuple[int, int]]]] = {
+    'strip_escseq': (strip_colors, 'sphinx.util.console.strip_colors', (9, 0)),
+}
+
+
+def __getattr__(name: str) -> Any:
+    if name not in _DEPRECATED_OBJECTS:
+        msg = f'module {__name__!r} has no attribute {name!r}'
+        raise AttributeError(msg)
+
+    from sphinx.deprecation import _deprecation_warning
+
+    deprecated_object, canonical_name, remove = _DEPRECATED_OBJECTS[name]
+    _deprecation_warning(__name__, name, canonical_name, remove=remove)
+    return deprecated_object
```

Comment:
Maybe it's time to extract a `sphinx.util` helper for this functionality?

Reply:
No, because you don't want to import this module (nor `sphinx.deprecation`) before you are using that object. And you don't want to have possible circular imports.

Reply:
Ok, makes sense - thanks for the explanation :+1: